pkg/proxy/proxier.go

```diff
@@ -368,12 +381,80 @@ func (proxier *Proxier) openOnePortal(portalIP net.IP, portalPort int, protocol
 	return nil
 }
 
+func (proxier *Proxier) lockPort(port int, owner ServicePortName) error {
```

Comment:
I don't think lock/unlock is the right analogy.  Claim/release.  Take/Put.  Something like that